src/Utils.cpp

```diff
@@ -44,6 +44,37 @@ std::optional<std::string> getAncestorPath(const std::string& path, const std::s
     return std::nullopt;
 }
 
+size_t pathDistance(const std::string& path1, const std::string& path2)
+{
+    std::filesystem::path p1(path1);
+    std::filesystem::path p2(path2);
+
+    auto distance = std::filesystem::relative(p1, p2);
+    return std::distance(distance.begin(), distance.end()) - 1;
+}
```

Comment:
I think you can set the arguments as a path directly rather than having to create another variable
```suggestion
size_t pathDistance(const std::filesystem::path& path1, const std::filesystem::path& path2)
{
    auto distance = std::filesystem::relative(path1, path2);
    return std::distance(distance.begin(), distance.end()) - 1;
}
```

Reply:
This is to be more in line with the other methods in the utils file.
```cpp
std::optional<std::string> getParentPath(const std::string& path);
std::optional<std::string> getAncestorPath(const std::string& path, const std::string& ancestorName);
size_t pathDistance(const std::string& path1, const std::string& path2);
std::string convertToScriptPath(const std::string& path);
```